src/components/tools/Charts/Shared/Annotation/annotationUtils.js

```diff
@@ -0,0 +1,311 @@
+    if (!arguments.length) return angle;
+    angle = Math.min(Math.max(_, 1e-6), Math.PI - 1e-6);
+    return render;
+  };
+
+  render.clockwise = function renderClockwise(_) {
+    if (!arguments.length) return clockwise;
+    clockwise = !!_;
+    return render;
+  };
+
+  render.x = function renderX(_) {
+    if (!arguments.length) return xValue;
+    xValue = _;
+    return render;
+  };
+
+  render.y = function renderY(_) {
+    if (!arguments.length) return yValue;
+    yValue = _;
+    return render;
+  };
+
+  return render;
+}
+
+/* --------------------------------------------
+ * straightArrow
+ *
+ *
+ */
+export function straightArrow() {
+  let xValue = (d) => d[0];
+  let yValue = (d) => d[1];
+
+  function render(data) {
+    data = data.map((d, i) => {
+      return [xValue.call(data, d, i), yValue.call(data, d, i)];
+    });
+    const path =
+      "M " +
+      data[0][0] +
+      "," +
+      data[0][1] +
+      "L " +
+      data[1][0] +
+      "," +
+      data[1][1];
+    return path;
+  }
+
+  render.x = function renderX(_) {
+    if (!arguments.length) return xValue;
+    xValue = _;
+    return render;
+  };
+
+  render.y = function renderY(_) {
+    if (!arguments.length) return yValue;
+    yValue = _;
+    return render;
+  };
+
+  return render;
+}
+
+/* --------------------------------------------
+ * straightLine
+ *
+ *
+ */
+export function threshold() {
+  let orientation = "v";
+  let extent = 100;
+  let xValue = (d) => d[0];
+  let yValue = (d) => d[1];
+
+  function render(data) {
+    data = data.map((d, i) => {
+      return [xValue.call(data, d, i), yValue.call(data, d, i)];
+    });
+
+    let path;
+    if (orientation === "v") {
+      path = "M" + data[0][0] + ",0" + orientation + extent;
+    } else {
+      path = "M0," + data[0][1] + orientation + extent;
+    }
+    return path;
+  }
+
+  render.orientation = function renderAngle(_) {
+    if (!arguments.length) return orientation;
+    orientation = _;
+    return render;
+  };
+
+  render.extent = function renderExtent(_) {
+    if (!arguments.length) return extent;
+    extent = _ + 10;
+    return render;
+  };
+
+  render.x = function renderX(_) {
+    if (!arguments.length) return xValue;
+    xValue = _;
+    return render;
+  };
+
+  render.y = function renderY(_) {
+    if (!arguments.length) return yValue;
+    yValue = _;
+    return render;
+  };
+
+  return render;
+}
+
+/* --------------------------------------------
+ * create some sample annotations
+ *
+ *
+ */
+export function getSampleAnnotations() {
+  return [
+    {
+      label: {
+        text: "1980",
+        position: {
+          top: "10%",
+          left: "20%",
+        },
+      },
+      connectors: [
+        {
+          type: "threshold",
+          orientation: "v",
+          source: {
+            anchor: "left-middle",
+            dx: 0,
+            dy: 10,
+          },
+          target: {},
+          style: { "stroke-dasharray": "6,4" },
```

Comment:
@clhenrick here's an example of how you'd pass in a custom style. this will override the default `stroke-dasharray` style of `0`

Reply:
That example is not setting a CSS custom property though. You would need to do something like:

```
style: {
  "--my-custom-prop": "some-value"
}
```

Is that how they are intended to be set?

Reply:
Nevermind, I'm seeing that the `$: setStyle` reactive function expression is prefixing the property names with "--". I think it could just be a regular function, not sure it needs to be reactive?

Reply:
The child components of the chart run twice on page load, I can't figure out why. The first time they run the `height` and `width` of the chart are not the same as the final values, so the labels & connectors get put in the wrong position when using `data` to position the elements. Making the functions reactive recalculated the positions with correct `height` & `width` values

Reply:
Got it, I didn’t even know you could make a function a reactive variable/statement in svelte.